### DISCUSSION TITLE
Interactive assessment of clinical heterogeneity in randomised controlled trials

### DISCUSSION THREAD

**Discussion by oliverodolin** [HTAhackathon2024/.github#24] [Ideas]
### What is the problem we're trying to solve

Trials need to be assessed to determine whether patient populations and trial designs are sufficiently homogenous to justify direct or indirect treatment comparisons (ITCs). ([Cochrane Handbook: Chapter 10](https://training.cochrane.org/handbook/current/chapter-10#section-10-10))
Certain aspects of the trial design, such as the selection criteria, are not easily visualised for decision-making. However, other aspects, like patient age and baseline disease severity, are more amenable to visualisation.

That's where this idea comes in!

A typical workflow is to extract this information into a large spreadsheet, and then present information about patient characteristics to clinical experts in a large table, for them to assess. It can be easy to miss outliers when data is presented in this way. Furthermore, clinical experts often do not have the time to spend carefully assessing the data in these tables.

My suggestion is to create a shiny app that would create simple, interactive visualisations of these data. (e.g., using box and whisker plots, where each trial's median age and age range are plotted side-by-side, and the user can mouse over a point to see the associated trial).

This could help facilitate decision-making by clinical experts, and also help reviewers/statisticians quickly explain and justify decisions to exclude particular studies from consideration in ITCs.

### How will we tackle the problem?

- [ ] Establish a standardised format for spreadsheet data, that we will import into R
- [ ] Write a function to load the data in the required format (plus do some simple sanity checks to ensure data were loaded correctly)
- [ ] Decide on some simple visualisations that would allow for quick assessment of clinical heterogeneity.
- [ ] Create some functions to produce these visualisations.
- [ ] Create a simple shiny application that would allow a user to upload a spreadsheet, and toggle through the visualisations for the different variables (eg., age, baseline severity) contained within that spreadsheet.

### Which tools will we use?

Mostly tidyverse packages. So ggplot2 for data viz, and dplyr for data wrangling (should be pretty straightforward wrangling, if we are specifying a required input format for the data).

### Any further resources

Here's an article that explains what clinical heterogeneity is, why it's important to assess, and how it was rarely assessed at the time of publication (2016): [Applicable or non-applicable: investigations of clinical heterogeneity in systematic reviews](https://doi.org/10.1186/s12874-016-0121-7)

**Comment by CKENNEDY-HE** [comment]
I think this is a great idea! Yes, I agree, often when looking at whether a trial is generalisable to the disease population of interest you usually look at tables so that it is not always to spot any outliers. Alongside trial baseline data it may also be good to have the possibility of incorporating registry data also.

**Comment by Neil-W1** [comment]
I would be interested in helping with this - have done some vaccine modelling work previously where homogeneity was assumed, so could be good to explore this further.

**Comment by TonyCroc** [comment]
This would be interesting, as we are looking to investigate homogeneity of data supplied from different sources eg clinicians , clinics
